### PR TITLE
version 0.4.0 doc

### PR DESCRIPTION
**Pull request by cojhal**
Added doc for 0.4.0, including report parameters.

### CODE REVIEW

**Comment by cojhal** [comment]
Is there official SAS documentation on report parameters that we can link to?

File: documentation/docs/api/SASReportPageElement.md [hunk 1]
```diff
@@ -44,3 +45,21 @@ Specify the name of the report page that you want to display. Either `pageName`
...
+If called before the element is added to the DOM, the promise will resolve
+after the page begins to load.
+
+A [ReportHandle](ReportHandle.md) are invalidated when attributes on the
+`SASReportPageElement` are changed and when the element is removed from the
+DOM. To obtain another [ReportHandle](ReportHandle.md), discard the previous
+result and call `getReportHandle` again.
+
+[ReportHandles](ReportHandle.md) from `SASReportPageElements` are never
+shared with other elements.
```
Comment (brmorr) [inline documentation/docs/api/SASReportPageElement.md:65]: Is this true?  I thought sas-report-page  acted more like sas-report-object in that it shares a report context with other objects/pages from the same report.

File: documentation/website/pages/en/versions.js [hunk 2]
```diff
@@ -55,9 +55,9 @@ function Versions(props) {
           <table className="versions">
             <tbody>
               {versions.map(
-                version =>
+                (version, i) =>
```
Comment (brmorr) [inline documentation/website/pages/en/versions.js:58]: Where did these changes come from?  It doesn't seem like you cut a new version, right?

File: documentation/docs/api/SASReportPageElement.md (hunk 1 above)
Comment (joylashley) [reply documentation/docs/api/SASReportPageElement.md:65]: There are a couple of sources for report parameters. "Working with Parameters in Reports" is here: https://documentation.sas.com/?cdcId=vacdc&cdcVersion=8.5&docsetId=vareportdata&docsetTarget=n1wv50n60ccq86n1nzp6zat1wj64.htm. 

And, "Using URL Parameters to View a Report" is here: https://documentation.sas.com/?cdcId=vacdc&cdcVersion=8.5&docsetId=vavwr&docsetTarget=p0l4zt68r3id4wn1fk3y3kconfg4.htm

Comment (joylashley) [reply documentation/docs/api/SASReportPageElement.md:65]: There are a couple of sources for report parameters. "Working with Parameters in Reports" is here: https://documentation.sas.com/?cdcId=vacdc&cdcVersion=8.5&docsetId=vareportdata&docsetTarget=n1wv50n60ccq86n1nzp6zat1wj64.htm. 

And, "Using URL Parameters to View a Report" is here: https://documentation.sas.com/?cdcId=vacdc&cdcVersion=8.5&docsetId=vavwr&docsetTarget=p0l4zt68r3id4wn1fk3y3kconfg4.htm.

File: documentation/website/pages/en/versions.js (hunk 2 above)
Comment (cojhal) [reply documentation/website/pages/en/versions.js:58]: This just fixed a warning from React when building the pages.
Correct, I have not done that yet.

Comment (brmorr) [reply documentation/website/pages/en/versions.js:58]: Great.  We'll do a separate changeset for the versioning after getting this change through.

File: documentation/docs/api/SASReportPageElement.md (hunk 1 above)
Comment (brmorr) [reply documentation/docs/api/SASReportPageElement.md:65]: My question was really for @cojhal, regarding the implementatio details in the sdk, and whether or not Page creates its own report context like sas-report, or uses a shared one like sas-report-object.  Given that there are multiple mentions of this throughout the new doc I'm guessing it is correct.  If so, great, the doc is correct, but I think we may want to rethink that as it doesn't seem like what the user would expect.  But that is an unrelated, post 0.4.0 discussion.

Comment (cojhal) [reply documentation/docs/api/SASReportPageElement.md:65]: @brmorr - I just ran some tests to be sure, and sas-report-page acts the way I thought it did. As it is currently implemented, page elements each get their own report context. Unlike report objects, there is also nothing stopping the user from creating two identical page elements.

File: documentation/docs/api/ReportHandle.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+[`SASReportPageElement`](SASReportPageElement.md), or
+[`SASReportObjectElement`](SASReportObjectElement.md).
+
+## Methods
+
+### setReportParameters(parameters: Object | undefined): void
+
+Set the state of all the parameters in a report. For information on how
+parameters can be added to reports, see the
+<a target="_blank" href="https://go.documentation.sas.com/?cdcId=vacdc&cdcVersion=8.4&docsetId=vareportdata&docsetTarget=n1wv50n60ccq86n1nzp6zat1wj64.htm">SAS documentation for parameters</a>.
```
Comment (cojhal) [inline documentation/docs/api/ReportHandle.md:18]: I linked to 8.4 to match the rest of the pages. Should we update them all to 8.5 instead?

Comment (brmorr) [reply documentation/docs/api/ReportHandle.md:18]: I changed all  the other doc to point to 8.5 last week

File: documentation/docs/api/ReportHandle.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
+---
+id: ReportHandle
+title: ReportHandle
+---
+
+A `ReportHandle` is used to control the state of a open report. A report's
```
Comment (brmorr) [inline documentation/docs/api/ReportHandle.md:6]:
```suggestion
A `ReportHandle` is used to control the state of an open report. A report's
```

Comment (brmorr) [reply documentation/docs/api/ReportHandle.md:6]: Other than this grammar fix, everything else LGTM.

Comment (brmorr) [reply documentation/docs/api/ReportHandle.md:6]: But as i mentioned before, we do need to update to 8.5 links.  Lets go ahead and just add that change to these changes.

File: documentation/docs/api/ReportHandle.md (hunk 3 above)
Comment (joylashley) [reply documentation/docs/api/ReportHandle.md:18]: Yes, it's best to link to 8.5. (We'll keep the same URLs when 8.5.1 ships in June.)

File: documentation/docs/api/ReportHandle.md (hunk 4 above)
Comment (joylashley) [reply documentation/docs/api/ReportHandle.md:6]: I agree, LGTM.

**Pull request merged**